File: 프로그래머스/KaKao/n진수 게임.swift
// 2018 KAKAO BLIND RECRUITMENT [3차] n진수 게임

func solution(_ n:Int, _ t:Int, _ m:Int, _ p:Int) -> String {
    var answer:String = ""
    var number:Int = 0
    var myIdx:Int = 1
    // 0, 1, 1, 0, 1, 1, 1, 0, 0
    while answer.count < t {
        for num in String(number, radix:n, uppercase: true) {
            if myIdx == p { // 내 차례가 되면
                answer += String(num)
            }
            
            if myIdx == m {
                myIdx = 0
            }
            
            if answer.count == t {
                break
            }
            
            myIdx += 1
        }
        number += 1
    }
    
    return answer
}

File: 프로그래머스/KaKao/n진수 게임.py
# 2018 KAKAO BLIND RECRUITMENT [3차] n진수 게임
def convert(num, base):
    tmp = "0123456789ABCDEF"
    q, r = divmod(num, base)

    if q == 0:
        return tmp[r]
    else:
        return convert(q, base) + tmp[r]


def solution(n, t, m, p):
    answer = ''
    newT = ""

    for i in range((m*t)+1):
        newT += convert(i, n)

    idx = p - 1

    while len(answer) < t:
        answer += newT[idx]
        idx += m

    return answer
